src/hostcalls.rs

```diff
@@ -0,0 +1,1071 @@
+                            header_data_ptr,
+                            string_body,
+                            trailer_data_ptr,
+                            timeout,
+                        ) {
+                            Some(expect_token) => expect_token,
+                            None => 0,
+                        };
+
+                        let return_token_add = mem.data_unchecked_mut().get_unchecked_mut(
+                            return_token as u32 as usize..return_token as u32 as usize + 4,
+                        );
+                        return_token_add.copy_from_slice(&token_id.to_le_bytes());
+
+                        println!("=>   proxy_http_call | upstream_data:  {}", string_upstream);
+                        println!(
+                            "                     | headers_data:   {:?}",
+                            deserialized_header
+                        );
+                        println!(
+                            "                     | body_data:      {}",
+                            string_body.unwrap_or("None")
+                        );
+                        println!(
+                            "                     | trailers_data:  {:?}",
+                            deserialized_trailer
+                        );
+                        println!("                     | timeout:        {:?}", timeout);
+                    }
+                    return Status::Ok as i32;
+                },
+            ))
+        }
+
+        "proxy_set_effective_context" => {
+            Some(Func::wrap(
+                &store,
+                |_caller: Caller<'_>, context_id: i32| -> i32 {
+                    // Default Function:
+                    // Expectation:
+                    println!("=>   proxy_set_effective_context | {}", context_id);
+                    return Status::Ok as i32;
+                },
+            ))
+        }
+
+        "proxy_done" => {
+            Some(Func::wrap(&store, |_caller: Caller<'_>| -> i32 {
+                // Default Function:
+                // Expectation:
+                println!("=>   proxy_done | ");
+                return Status::InternalFailure as i32;
+            }))
+        }
+
+        // Host function implementation template
+        // "fn_name" => {
+        //     Some(Func::wrap(
+        //         &store, |_caller: Caller<'_>,
+        //                  _args: i32| -> i32 {
+        //             println!("=>   proxy_fn_name | ");
+        //             return Status::InternalFailure as i32;
+        //         }
+        //     ))
+        // },
+        _ => None,
+    }
+}
+
+pub mod serial_utils {
+
+    type Bytes = Vec<u8>;
+    use rand::Rng;
+    use std::convert::TryFrom;
+
+    const CHARSET: &[u8] = b"ABCDEFGHIJKLMNOPQRSTUVWXYZ\
+                            abcdefghijklmnopqrstuvwxyz\
+                            0123456789)(*&^%$#@!~";
+
+    pub fn serialize_property_path(path: Vec<&str>) -> Bytes {
```

Comment:
```
warning: function is never used: `serialize_property_path`
    --> src/hostcalls.rs:1000:12
     |
1000 |     pub fn serialize_property_path(path: Vec<&str>) -> Bytes {
     |            ^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: `#[warn(dead_code)]` on by default
```

Reply:
I think this function will eventually be used when the host environment is complete.